### DISCUSSION TITLE
Cannot install Helix on Master ( Server does not allow request for unadvertised object 793d58266924e6efcc40e411663393e9d72bec87)

### DISCUSSION THREAD

**Discussion by sezna** [helix-editor/helix#8612] [Troubleshooting] [answered]
### Summary

Hello!

Happy user of Helix here. I wanted to update my build to the latest. I typically install from master with `cargo`. On the latest build, I get the following error text:

```
error: failed to run custom build command for `helix-term v0.6.0 (/home/_/helix/helix-term)`

Caused by:
  process didn't exit successfully: `/home/_/helix/target/release/build/helix-term-4821aa082921f959/build-script-build` (exit status: 101)
  --- stdout
  Fetching 163 grammars
  162 up to date git grammars
  Failure 1/1: wren Git command failed.
  Stdout:
  Stderr: error: Server does not allow request for unadvertised object 793d58266924e6efcc40e411663393e9d72bec87


  --- stderr
  thread 'main' panicked at 'Failed to fetch tree-sitter grammars: 1 grammars failed to fetch', helix-term/build.rs:5:26
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
warning: build failed, waiting for other jobs to finish...
error: failed to compile `helix-term v0.6.0 (/home/_/helix/helix-term)`, intermediate artifacts can be found at
```

### Reproduction Steps

 

I read that this could be caused by submodules, so I sync'd submodules, but no dice.
- `git submodule sync; git submodule update --force --recursive --init --remote`

I thought perhaps it is related to some system level git weirdness, so I tried building with `CARGO_NET_GIT_FETCH_WITH_CLI=1`, but still no dice.

I also tried `cargo clean && cargo install`.

Unfortunately, none of these helped. The command I have been trying to use to install is:

`cargo install --path helix-term --locked`

### Helix log

Helix log is N/A for this case.

### Platform

WSL

### Terminal Emulator

Windows Terminal

### Helix Version

latest master, hash `68c7537de`

**Comment by pascalkuthe** [comment]
what is your git version?

**Comment by sezna** [comment]
Hello!

```
$> git --version
git version 2.25.1
```

Thanks for your help!

**Reply by the-mikedavis** [reply, marked as answer]
2.25 is a bit old. That version might not have support for cloning at exact revisions with sourcehut (github seems to be fine though, oddly). Try updating your git version and rebuilding.

**Reply by sezna** [reply]
This did it. I updated to 2.42 and it worked. Thanks!
